### DISCUSSION TITLE
Nix Flake?

### DISCUSSION THREAD

**Discussion by s1syph0s** [esp-rs/espup#428] [General]
Hi,

I am using NixOS, and wanted to try to install the esp rust toolchain. Using espup can work in NixOS, but I think that it could be great, if we can install the esp toolchains using nix flakes. Is there a plan to support installing the toolchains using nix?

**Comment by SergioGasquez** [comment]
Hi! 

I don't think there is a plan for adding a nix flake from our side, but if you end up doing one or you find one that works, it would be very valuable for others. I know there are some Nix users in https://matrix.to/#/#esp-rs:matrix.org

**Reply by SergioGasquez** [reply]
Here are some resources/issues related to Nix/NixOS that I've gathered from the Matrix channel: 
- https://github.com/esp-rs/esp-idf-template/issues/64#issuecomment-1308878695
- https://github.com/esp-rs/esp-idf-sys/issues/184
- https://github.com/thiskappaisgrey/nixpkgs-esp-dev-rust/tree/rust/shells
